docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc

```diff
@@ -1,21 +1,331 @@
 [[analysis-delimited-payload-tokenfilter]]
-=== Delimited Payload Token Filter
-
-Named `delimited_payload`. Splits tokens into tokens and payload whenever a delimiter character is found.
+=== Delimited payload token filter
+++++
+<titleabbrev>Delimited payload</titleabbrev>
+++++
 
 [WARNING]
-============================================
+====
+The older name `delimited_payload_filter` is deprecated and should not be used
+with new indices. Use `delimited_payload` instead.
+====
+
+Separates a token stream into tokens and payloads based on a specified
+delimiter.
+
+For example, you can use the `delimited_payload` filter with a `|` delimiter to
+split `the|1 quick|2 fox|3` into the tokens `the`, `quick`, and `fox`
+with respective payloads of `1`, `2`, and `3`.
+
+This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/payloads/DelimitedPayloadTokenFilter.html[DelimitedPayloadTokenFilter].
+
+[NOTE]
+.Payloads
+====
+A payload is user-defined binary data associated with a token position and
+stored as base64-encoded bytes. Payloads are often used with the
+<<query-dsl-script-score-query,`script_score`>> query to calculate custom scores
+for documents during a search.
+
+{es} does not store token payloads by default. To store payloads, you must:
+
+
+* Set mapping parameters as follows for any field storing payloads:
+** <<mapping-store,`store`>> to `true`
+** <<term-vector,`term_vector`>> to `with_positions_payloads` or
+   `with_positions_offsets_payloads`
+
+* Use an index analyzer that includes the `delimted_payload` filter
+====
```

Comment:
I am not sure about the correctness of this paragraph.  So, for a text/keyword field ES can create 3 Lucene fields:  1) usual indexed field broken into terms and used for search. Here we don't sore payloads (I think) and definitely never use them in any search query. 2) a stored field with index option `store:true`.  But I am not sure why you mentioned it here (it seems to me that it doesn't have to do anything with payloads).  3) a term vectors field with index option of `term_vector`. these are primarily used for highlighting, but can be used just for retrieval purpose as well (as in your examples). And here we can store payloads but again we don't use payloads for anything intelligent besides just retrieval. 

nit: `delimted_payload` -> `delimited_payload`

Reply:
Thanks for another great catch. I've removed references to the `store` requirement throughout. I also fixed the typo.